### PR TITLE
fix(web): fixed a couple variant issues

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

Fixed a couple of issues that I've found in the Variants project:
- the appropriate variant label in the editor
- the loading state for the variants list page
- added tenant filter to the conditions
- variant template mapping

### Other information (Screenshots)

https://github.com/novuhq/novu/assets/2607232/84118ff4-e077-43c8-b594-c7449f7a33ca

Screenshot 2023-10-26 at 18 25 10

Screenshot 2023-10-26 at 18 25 20

Screenshot 2023-10-26 at 18 26 25

### CODE REVIEW

File: apps/web/src/pages/templates/components/StepNameLabel.tsx [hunk 1]
```diff
@@ -3,16 +3,19 @@ import { Text, colors, Variant, VariantsFile } from '@novu/design-system';
...
+  const { variantsCount } = useStepVariantsCount();
 
   if (isUnderTheStepPath) {
     return null;
   }
 
-  const variantNumber = variantIndex ? variantIndex + 1 : 1;
+  const variantArrayIndex = variantIndex ? variantIndex + 1 : 1;
+  const variantNumber = variantsCount - variantArrayIndex + 1;
   const text = isUnderVariantPath ? `Variant ${variantNumber}` : `${variantsCount} variants`;
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/StepNameLabel.tsx:19]: fixed the variant label under the variant name in the editor

File: apps/web/src/pages/templates/components/VariantsPage.tsx [hunk 2]
```diff
@@ -75,7 +77,9 @@ export function VariantsPage() {
             />
           );
         })}
-        <VariantItemCard isReadonly={isReadonly} key={stepUuid} variant={step} nodeType="variantRoot" />
+        {step && variants.length > 0 && (
+          <VariantItemCard isReadonly={isReadonly} key={stepUuid} variant={step} nodeType="variantRoot" />
+        )}
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/VariantsPage.tsx:82]: when no variants and no step don't render the root step in the variants list sidebar

File: apps/web/src/pages/templates/components/templateToFormMappers.ts [hunk 3]
```diff
@@ -223,47 +224,53 @@ export const mapNotificationTemplateToForm = (template: INotificationTemplate):
...
+      template,
+      metadata: mapFormStepDelayMetadata(formStep),
+    };
+  }
+
+  return {
+    ...step,
+    ...(!!('variants' in step) && {
+      variants: step.variants?.map((variant) => mapFormStepToDto(variant)) ?? ([] as any),
+    }),
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/templateToFormMappers.ts:267]: map form variants to appropriate dto objects

File: apps/web/src/pages/templates/hooks/useFilterPartsList.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ const FILTER_PART_LIST: IFilterTypeList[] = [
     value: FilterPartTypeEnum.SUBSCRIBER,
     label: FILTER_TO_LABEL[FilterPartTypeEnum.SUBSCRIBER],
   },
+  { value: FilterPartTypeEnum.TENANT, label: FILTER_TO_LABEL[FilterPartTypeEnum.TENANT] },
```
Comment (LetItRock) [inline apps/web/src/pages/templates/hooks/useFilterPartsList.ts:14]: added tenant filter to conditions component

**Review by ainouzgali — APPROVED** [review]

**Pull request merged**